promoter/promoter.go

```diff
@@ -0,0 +1,92 @@
+package promoter
+
+import (
+	"context"
+	"sync"
+
+	"github.com/sirupsen/logrus"
+	"gitlab.com/SkynetLabs/skyd/node/api/client"
+	"go.mongodb.org/mongo-driver/mongo"
+)
+
+type (
+	// Health contains health information about the promoter. Namely the
+	// database and skyd. If everything is ok all fields are 'nil'.
+	// Otherwise the corresponding fields will contain an error.
+	Health struct {
+		Database error
+		Skyd     error
+	}
+
+	// Promoter is a wrapper around a skyd and a database client. It makes
+	// sure that skyd watches all the siacoin addresses it is supposed to
+	// and is capable of adding new addresses to watch and removing old
+	// ones. It can also track the incoming funds that users have sent to
+	// their assigned addresses.
+	Promoter struct {
+		staticClient              *mongo.Client
+		staticDB                  *mongo.Database
+		staticLogger              *logrus.Entry
+		staticColWatchedAddresses *mongo.Collection
```

Comment:
nit: Keeping a collection on the struct improves performance a tiny bit because it saves some code execution. That's why I did it in `accounts`. The issue is that with time is gets annoying and inflexible. That's why in `pinner` I'm always using `staticDB.Collection(colSkylinks).Find()`.
Both work well and the difference is small, I just wanted to share that I naturally moved from one to the other.